### ISSUE TITLE
SQLiteDatabase.updateWithOnConflict: IllegalArgumentException: Empty values

### ISSUE THREAD

**Issue by AndyScherzinger** [nextcloud/android#425]
Via Google Play, 1.4.0 RC only (any of them RC1-RC4)!

```
java.lang.IllegalArgumentException: Empty values
	at android.database.sqlite.SQLiteDatabase.updateWithOnConflict(SQLiteDatabase.java:1680)
	at android.database.sqlite.SQLiteDatabase.update(SQLiteDatabase.java:1660)
	at com.owncloud.android.providers.FileContentProvider.updateFilesTableAccordingToShareInsertion(FileContentProvider.java:349)
	at com.owncloud.android.providers.FileContentProvider.insert(FileContentProvider.java:283)
	at com.owncloud.android.providers.FileContentProvider.insert(FileContentProvider.java:227)
	at android.content.ContentProviderOperation.apply(ContentProviderOperation.java:245)
	at com.owncloud.android.providers.FileContentProvider.applyBatch(FileContentProvider.java:549)
	at android.content.ContentProvider$Transport.applyBatch(ContentProvider.java:301)
	at android.content.ContentProviderClient.applyBatch(ContentProviderClient.java:402)
	at android.content.ContentResolver.applyBatch(ContentResolver.java:1297)
	at com.owncloud.android.datamodel.FileDataStorageManager.saveSharesInFolder(FileDataStorageManager.java:1484)
	at com.owncloud.android.operations.RefreshFolderOperation.refreshSharesForFolder(RefreshFolderOperation.java:504)
	at com.owncloud.android.operations.RefreshFolderOperation.run(RefreshFolderOperation.java:216)
	at com.owncloud.android.lib.common.operations.RemoteOperation.run(RemoteOperation.java:303)
	at java.lang.Thread.run(Thread.java:818)
```

Comments:
1.4.0 RC4: Opening one folder crashes the app
1.4.0 RC2: Nextcloud app beendet sich sobald ich einen Ordner öffnen der geshared ist

**Comment by eppfel** [comment]
Two crashes, same Error log, looks similar to yours:

1. In app switcher, tapped the icon in the header, to get to the android app settings for nextcloud
2. Deleted the data and started the app; Opening still shows folder structure then crashes 

```log
************ CAUSE OF ERROR ************

java.lang.IllegalArgumentException: Empty values
	at android.database.sqlite.SQLiteDatabase.updateWithOnConflict(SQLiteDatabase.java:1542)
	at android.database.sqlite.SQLiteDatabase.update(SQLiteDatabase.java:1522)
	at com.owncloud.android.providers.FileContentProvider.updateFilesTableAccordingToShareInsertion(FileContentProvider.java:349)
	at com.owncloud.android.providers.FileContentProvider.insert(FileContentProvider.java:283)
	at com.owncloud.android.providers.FileContentProvider.insert(FileContentProvider.java:227)
	at android.content.ContentProviderOperation.apply(ContentProviderOperation.java:302)
	at com.owncloud.android.providers.FileContentProvider.applyBatch(FileContentProvider.java:549)
	at android.content.ContentProvider$Transport.applyBatch(ContentProvider.java:315)
	at android.content.ContentProviderClient.applyBatch(ContentProviderClient.java:419)
	at android.content.ContentResolver.applyBatch(ContentResolver.java:1279)
	at com.owncloud.android.datamodel.FileDataStorageManager.saveSharesInFolder(FileDataStorageManager.java:1484)
	at com.owncloud.android.operations.RefreshFolderOperation.refreshSharesForFolder(RefreshFolderOperation.java:504)
	at com.owncloud.android.operations.RefreshFolderOperation.run(RefreshFolderOperation.java:216)
	at com.owncloud.android.lib.common.operations.RemoteOperation.run(RemoteOperation.java:303)
	at java.lang.Thread.run(Thread.java:818)

************ DEVICE INFORMATION ***********
Brand: Sony
Device: E5823
Model: E5823
Id: ***removed***
Product: E5823

************ FIRMWARE ************
SDK: 23
Release: 6.0.1
Incremental: 724807262
```

**Comment by realitygaps** [comment]
I get this similar crash on latest fdroid beta:

************ CAUSE OF ERROR ************

java.lang.IllegalArgumentException: Empty values
	at android.database.sqlite.SQLiteDatabase.updateWithOnConflict(SQLiteDatabase.java:1545)
	at android.database.sqlite.SQLiteDatabase.update(SQLiteDatabase.java:1525)
	at com.owncloud.android.providers.FileContentProvider.updateFilesTableAccordingToShareInsertion(FileContentProvider.java:349)
	at com.owncloud.android.providers.FileContentProvider.insert(FileContentProvider.java:283)
	at com.owncloud.android.providers.FileContentProvider.insert(FileContentProvider.java:227)
	at android.content.ContentProviderOperation.apply(ContentProviderOperation.java:302)
	at com.owncloud.android.providers.FileContentProvider.applyBatch(FileContentProvider.java:549)
	at android.content.ContentProvider$Transport.applyBatch(ContentProvider.java:316)
	at android.content.ContentProviderClient.applyBatch(ContentProviderClient.java:438)
	at android.content.ContentResolver.applyBatch(ContentResolver.java:1318)
	at com.owncloud.android.datamodel.FileDataStorageManager.saveSharesInFolder(FileDataStorageManager.java:1484)
	at com.owncloud.android.operations.RefreshFolderOperation.refreshSharesForFolder(RefreshFolderOperation.java:504)
	at com.owncloud.android.operations.RefreshFolderOperation.run(RefreshFolderOperation.java:216)
	at com.owncloud.android.lib.common.operations.RemoteOperation.run(RemoteOperation.java:303)
	at java.lang.Thread.run(Thread.java:761)

************ DEVICE INFORMATION ***********
Brand: Motorola
Device: athene
Model: Moto G4
Id: NMF26Q
Product: cm_athene

************ FIRMWARE ************
SDK: 25
Release: 7.1.1
Incremental: c89318e12a

**Comment by AndyScherzinger** [comment]
Fixed via #492
